tests/pass/tree-borrows/transmute-unsafecell.rs

```diff
@@ -1,13 +1,32 @@
 //@compile-flags: -Zmiri-tree-borrows
 
+//! Testing `mem::transmute` between types with and without interior mutability.
+//! All transmutations should work, as long as we don't do any actual accesses
+//! that violate immutability.
+
 use core::cell::UnsafeCell;
 use core::mem;
 
 fn main() {
     unsafe {
-        let x = &0i32;
-        // As long as we only read, transmuting this to UnsafeCell should be fine.
-        let cell_x: &UnsafeCell<i32> = mem::transmute(&x);
```

Comment:
O_o this was using a transmute on `&&i32` to `&UnsafeCell<i32>` whoops

Reply:
No comment. That's what the `assert!` was added for ;)